Sources/RoutingKit/TrieRouter/TrieRouterBuilder.swift

```diff
@@ -119,6 +119,9 @@ public struct TrieRouterBuilder<Output: Sendable>: RouterBuilder {
             )
             return node.copyWith(wildcard: newWildcard)
         case .partialParameter(let template, let components, let parameters):
+            guard parameters.count == Set(parameters).count else {
+                preconditionFailure("Partial \":\(template)\" contains multiple same-named parameters")
+            }
```

Comment:
I dislike that we crash for this, but I also don't see any good alternatives.

Meanwhile, why not:

```suggestion
           precondition(parameters.count == Set(parameters).count, #"Partial ":\#(template)" contains multiple parameters with the same name"#)
```